File: comms/bulk-messaging/sender-pools/list-sender-pool-operations/list-sender-pool-operations.py
from twilio import TwilioComms
import datetime

client = TwilioComms(
    account_id="<username>",
    auth_token="<password>",
)

client.sender_pools.list_operations(
    start_date=datetime.datetime.fromisoformat("2024-01-15T09:30:00+00:00"),
    end_date=datetime.datetime.fromisoformat("2024-01-15T09:30:00+00:00"),
    status="PROCESSING",
    page_token="pageToken",
    page_size=50,
)

File: comms/bulk-messaging/sender-pools/list-sender-pool-operations/list-sender-pool-operations.cs
using Twilio.Comms;
using System.Threading.Tasks;
using System;
using System.Globalization;

public partial class Examples
{
    public async Task Example() {
        var client = new TwilioComms(
            accountId: "TWILIO_ACCOUNT_SID",
            authToken: "TWILIO_AUTH_TOKEN"
        );

        await client.SenderPools.ListOperationsAsync(
            new SenderPoolsListOperationsRequest {
                StartDate = DateTime.Parse("2024-01-15T09:30:00Z", null, DateTimeStyles.AdjustToUniversal),
                EndDate = DateTime.Parse("2024-01-15T09:30:00Z", null, DateTimeStyles.AdjustToUniversal),
                Status = OperationStatus.Processing,
                PageToken = "pageToken",
                PageSize = 50
            }
        );
    }

}
